core/shared/platform/common/posix/posix_time.c

```diff
@@ -15,3 +16,142 @@ os_time_get_boot_microsecond()
 
     return ((uint64)ts.tv_sec) * 1000 * 1000 + ((uint64)ts.tv_nsec) / 1000;
 }
+
+uint64
+os_clock_res_get(bh_clock_id_t clock_id, uint64 *resolution)
+{
+    #ifdef BH_PLATFORM_WINDOWS 
+    switch (clock_id) {
+        case BH_CLOCK_MONOTONIC:
+        {
+            const uint64 result = (uint64)NANOSECONDS_PER_SECOND
+                                  / calculate_monotonic_clock_frequency();
+            *resolution = result;
+            return BHT_OK;
+        }
+        case BH_CLOCK_REALTIME:
+        case BH_CLOCK_PROCESS_CPUTIME_ID:
+        case BH_CLOCK_THREAD_CPUTIME_ID:
+        {
+            PULONG MaximumTime;
+            PULONG MinimumTime;
+            PULONG CurrentTime;
+            NTSTATUS
+            NTAPI Res;
+            if (Res = NtQueryTimerResolution(&MaximumTime, &MinimumTime,
+                                             &CurrentTime)) {
+                !NT_SUCCESS(Res);
+            }
+            else
+
+            {
+                return NT_ERROR(Res);
+            }
+        }
+    }
+
+#else {
+    __wasi_clockid_t nclock_id;
+        if (!convert_clockid(clock_id, &nclock_id))
+            return __WASI_EINVAL;
+        struct timespec ts;
+        if (clock_getres(nclock_id, &ts) < 0)
+            return convert_errno(errno);
```

Comment:
I'd suggest returning `BHT_ERROR` here, and then do `convert_errno()` in the `posix.c` file - I don't think `convert_errno` is available in this file.